public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,903 @@
+  concentrates exclusively on the reactive forms techniques. For 
+  information on _template driven forms_, see the [Template Guide](forms.html).
+
+  You'll learn about reactive forms by building one from scratch.
+
+  In the next section, you'll prepare your project for reactive forms.
+  Then you'll [learn about the Angular form classes](#essentials) and how to use them in a reactive form.
+
+.l-main-section
+a#setup
+:marked
+  ## Setup
+
+  Follow the steps in the [_Setup_ guide](../setup.html "Setup guide") 
+  for creating a new project folder (perhaps called `reactive-forms`)
+  based on the _QuickStart seed_.
+
+.l-main-section
+a#data-model
+:marked
+  ## Create a data model
+  The focus of this guide is a reactive forms component that edits a hero.
+  You'll need a `hero` class and some hero data.
+  Create a new `data-model.ts` file in the `app` directory and copy the content below into it.
+
++makeExample('reactive-forms/ts/app/data-model.ts', '','app/data-model.ts')(format=".")
+
+:marked
+  The file exports two classes and two constants. The `Address` 
+  and `Hero` classes define the application _data model_. 
+  The `heroes` and `states` constants supply the test data. 
+
+.l-main-section
+a#create-component
+:marked
+  ## Create a _reactive forms_ component
+  Make a new file called 
+  `hero-detail.component.ts` in the `app` directory and import these symbols:
+
++makeExample('reactive-forms/ts/app/hero-detail-1.component.ts', 'imports','app/hero-detail.component.ts')(format=".")
+
+:marked
+  Now enter the `@Component` decorator that specifies the `HeroDetailComponent` metadata:
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'metadata','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  The `moduleId: module.id` lets you use 
+  [component-relative paths](./cookbook/component-relative-paths.html) in file URLs 
+  such as when specifying the `templateUrl`.
+
+  Next, create an exported `HeroDetailComponent` class with a `FormControl`. 
+  Notice that the `FormControl` is within a `FormGroup`. While you don't 
+  _have_ to put a `FormControl` in a `FormGroup` for it to work, forms 
+  usually contain multiple controls and grouping them offers advantages 
+  you'll learn later in this guide.
+
++makeExample('reactive-forms/ts/app/hero-detail-1.component.ts', 'v1','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Here you are creating a `FormGroup` called `heroForm` with a child `FormControl` called `name`.
+  It will be bound in the template to an HTML `input` box for the hero name. 
+
+  A `FormControl` constructor accepts three, optional arguments: 
+  the initial data value, an array of validators, and an array of async validators.
+  
+  This simple control doesn't have data or validators.
+  In real apps, most form controls have both. 
+
+.l-sub-section
+  :marked
+    Validators aren't covered at all in this guide.
```

Comment:
I think it would be simple enough to add `Validators.required` somewhere in the guide to illustrate that validators are just functions that can be passed in the class (rather than validator directives, like in TD). It's an important point to make.